docs/API.md

```diff
@@ -951,6 +1008,217 @@ __Example__
+
+__Parameters__
+
+|Param   | Type	  | Description  |
+|:--- |:--- |:--- |
+| ``bucketName``  | _String_ | Name of the bucket.  |
+| ``config``  | _ObjectLockConfiguration_ | Object lock Configuration  |
+
+
+| Return Type	  | Exceptions	  |
+|:--- |:--- |
+| ``None``  | Listed Exceptions: |
+|        |  ``InvalidBucketNameException`` : upon invalid bucket name. |
+|        |  ``RegionConflictException`` : upon  passed region conflicts with the one previously specified. |
+|        | ``NoSuchAlgorithmException`` : upon requested algorithm was not found during signature calculation.           |
+|        | ``InsufficientDataException`` : Thrown to indicate that reading given InputStream gets EOFException before reading given length. |
+|        | ``IOException`` : upon connection error.            |
+|        | ``InvalidKeyException`` : upon an invalid access key or secret key.           |
+|        | ``NoResponseException`` : upon no response from server.            |
+|        | ``org.xmlpull.v1.XmlPullParserException`` : upon parsing response XML.            |
+|        | ``ErrorResponseException`` : upon unsuccessful execution.            |
+|        | ``InternalException`` : upon internal library error.        |
+|        | ``InvalidResponseException`` : upon a non-xml response from server.        |
+
+__Example__
+
+
+```java
+try {
+      // Create bucket if it doesn't exist.
+      boolean found = s3Client.bucketExists("my-bucketname");
+      if (found) {
+           System.out.println("my-bucketname already exists");
+         } else {
+           // Create bucket 'my-bucketname' with object lock functionality enabled
+           s3Client.makeBucket("my-bucketname", null, true);
+           System.out.println("my-bucketname is created successfully with object lock functionality enabled.");
+         }
+   
+         // Declaring config with Retention mode as Compliance and
+         // duration as 100 days
+         ObjectLockConfiguration setConfig = new ObjectLockConfiguration(RetentionMode.COMPLIANCE,100 , DurationUnit.DAYS);
+   
+         // Set object lock configuration
+         s3Client.setDefaultRetention("my-bucketname",config);
+  
+} catch (MinioException e) {
+  System.out.println("Error occurred: " + e);
+}
+```
+
+
+<a name="getDefaultRetention"></a>
+### getDefaultRetention(String bucketName)
+`public void getDefaultRetention(String bucketName)`
+
+Get default retention of bucket.
+   
+[View Javadoc](http://minio.github.io/minio-java/io/minio/MinioClient.html#getDefaultRetention-java.lang.String- )
+
+__Parameters__
+
+|Param   | Type	  | Description  |
+|:--- |:--- |:--- |
+| ``bucketName``  | _String_ | Name of the bucket.  |
+
+
+
+| Return Type	  | Exceptions	  |
+|:--- |:--- |
+| ``None``  | Listed Exceptions: |
+|        |  ``InvalidBucketNameException`` : upon invalid bucket name. |
+|        |  ``RegionConflictException`` : upon  passed region conflicts with the one previously specified. |
```

Comment:
Instead of RegionConflictException you should talk about InvalidArgumentException as that is what is being thrown in the code.